tests/framework/i18n/FallbackMessageFormatterTest.php

```diff
@@ -52,6 +59,38 @@ public function patterns()
                 ]
             ],
 
+            [
+                'Here is a big number: {'.self::F.', number}', // pattern
+                'Here is a big number: '.self::F_VALUE_FORMATTED, // expected
+                [ // params
+                    self::F => self::F_VALUE
+                ]
+            ],
+
+            [
+                'Here is a big number: {'.self::F.', number, integer}', // pattern
+                'Here is a big number: '.self::F_VALUE_FORMATTED, // expected
+                [ // params
+                    self::F => self::F_VALUE
+                ]
+            ],
+
+            [
+                'Here is a big number: {'.self::D.', number}', // pattern
+                'Here is a big number: '.self::D_VALUE_FORMATTED, // expected
+                [ // params
+                    self::D => self::D_VALUE
+                ]
+            ],
+
+            [
+                'Here is a big number: {'.self::D.', number, integer}', // pattern
+                'Here is a big number: '.self::D_VALUE_FORMATTED_INTEGER, // expected
+                [ // params
+                    self::D => self::D_VALUE
+                ]
+            ],
```

Comment:
also we need negative tests for fallback

`Here is a percent: {p, number, percent}` throws expected exception